dev-test/githunt/typed-document-nodes.ts

```diff
@@ -1,8 +1,8 @@
 import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
 export type Maybe<T> = T | null;
 export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
-export type MakeOptional<T, K extends keyof T> = Omit<T, K> &
-  { [SubKey in keyof Pick<T, K>]?: Maybe<Pick<T, K>[SubKey]> };
+export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
```

Comment:
I think we can experiment with TS generics inference here.
Something like :

```ts
export type MakeMaybe<T, K extends keyof T, AndOptional extends boolean> = AndOptional extends true ? Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> } : Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> ;
```

What do you think?